python-sdk/nuscenes/eval/tracking/loaders.py

```diff
@@ -0,0 +1,154 @@
+from bisect import bisect
+from typing import List, Dict
+
+import numpy as np
+from pyquaternion import Quaternion
+
+from nuscenes.eval.common.data_classes import EvalBoxes
+from nuscenes.utils.splits import create_splits_scenes
+from nuscenes.eval.tracking.data_classes import TrackingBox
+from nuscenes.nuscenes import NuScenes
+
+
+def interpolate_tracking_boxes(left_box: TrackingBox, right_box: TrackingBox, right_ratio: float) -> TrackingBox:
+    """
+    Linearly interpolate box parameters between two boxes.
+    :param left_box: A Trackingbox.
+    :param right_box: Another TrackingBox
+    :param right_ratio: Weight given to the right box.
+    :return: The interpolated TrackingBox.
+    """
+    def interp_list(left, right, rratio):
+        return tuple(
+            (1.0 - rratio) * np.array(left, dtype=float)
+            + rratio * np.array(right, dtype=float)
+        )
+
+    def interp_float(left, right, rratio):
+        return (1.0 - rratio) * float(left) + rratio * float(right)
+
+    # Interpolate quaternion.
+    rotation = Quaternion.slerp(
+        q0=Quaternion(left_box.rotation),
+        q1=Quaternion(right_box.rotation),
+        amount=right_ratio
+    ).elements
+
+    # Score will remain -1 for GT.
+    tracking_score = interp_float(left_box.tracking_score, right_box.tracking_score, right_ratio)
+
+    return TrackingBox(sample_token=right_box.sample_token,
+                       translation=interp_list(left_box.translation, right_box.translation, right_ratio),
+                       size=interp_list(left_box.size, right_box.size, right_ratio),
+                       rotation=rotation,
+                       velocity=interp_list(left_box.velocity, right_box.velocity, right_ratio),
+                       ego_dist=interp_float(left_box.ego_dist, right_box.ego_dist, right_ratio),  # May be inaccurate.
+                       tracking_id=right_box.tracking_id,
+                       tracking_name=right_box.tracking_name,
+                       tracking_score=tracking_score)
+
+
+def interpolate_tracks(tracks_by_timestamp: Dict[int, List[TrackingBox]]) -> Dict[int, List[TrackingBox]]:
+    """
+    Interpolate the tracks to fill in holes, especially since GT boxes with 0 lidar points are removed.
+    We are rediscovering an information that already exists in nuscenes.
```

Comment:
@cdicle-nutonomy @holger-nutonomy : I don't understand this line. Can you rephrase? There is also a typo: it should be "information" rather than "an information".

Reply:
@cdicle-nutonomy

Reply:
Removing this line.